source/extensions/quic_listeners/quiche/envoy_quic_dispatcher.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#pragma GCC diagnostic push
+// QUICHE allows unused parameters.
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+// QUICHE uses offsetof().
+#pragma GCC diagnostic ignored "-Winvalid-offsetof"
+#pragma GCC diagnostic ignored "-Wtype-limits"
+
+#include "quiche/quic/core/quic_dispatcher.h"
+#include "quiche/quic/core/quic_utils.h"
+
+#pragma GCC diagnostic pop
+
+#include <string>
+
+#include "envoy/network/listener.h"
+#include "server/connection_handler_impl.h"
+
+namespace Envoy {
+namespace Quic {
+
+// Envoy specific provider of server connection id and decision maker of
+// accepting new connection or not.
+class EnvoyQuicCryptoServerStreamHelper : public quic::QuicCryptoServerStream::Helper {
+public:
+  ~EnvoyQuicCryptoServerStreamHelper() override = default;
+
+  // quic::QuicCryptoServerStream::Helper
+  quic::QuicConnectionId
+  GenerateConnectionIdForReject(quic::QuicTransportVersion /*version*/,
+                                quic::QuicConnectionId /*connection_id*/) const override {
+    // TODO(danzh): create reject connection id based on given connection_id.
```

Comment:
just fyi, this function is no longer used for rejects. I'll take an action item to refactor it.